staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_discovery_controller.go

```diff
@@ -213,6 +213,29 @@ func (c *DiscoveryController) Run(stopCh <-chan struct{}) {
 		return
 	}
 
+	// initially sync all group versions to make sure we serve complete discovery
+	if err := wait.PollImmediateUntil(time.Second, func() (bool, error) {
+		crds, err := c.crdLister.List(labels.Everything())
+		if err != nil {
+			utilruntime.HandleError(fmt.Errorf("failed to initially list CRDs: %v", err))
+			return false, nil
+		}
+		for _, crd := range crds {
+			for _, v := range crd.Spec.Versions {
+				gv := schema.GroupVersion{crd.Spec.Group, v.Name}
+				if err := c.sync(gv); err != nil {
+					utilruntime.HandleError(fmt.Errorf("failed to initially sync CRD version %v: %v", gv, err))
+					return false, nil
+				}
+			}
+		}
+		return true, nil
+	}, stopCh); err != nil {
+		utilruntime.HandleError(fmt.Errorf("timed out waiting for discovery endpoint to initialize"))
```

Comment:
panic on these please.  Based on the code it should never happen, but I don't want to ever silently *not* run this.

Reply:
ok, timeout happens only when stopCh closed, so we can simply error